src/main/ruby/core/io.rb

```diff
@@ -2581,7 +2581,8 @@ def write_nonblock(data)
     @ibuffer.unseek!(self) unless @sync
 
     self.nonblock = true
-    Truffle.invoke_primitive :io_write_nonblock, self, data
+    Truffle::POSIX.write_rope(self.descriptor, data, true)
+    data.bytesize
```

Comment:
The return value of a write operation should be the number of bytes written, which can differ with the bytesize, so I think write_rope should return the number of bytes it wrote.

Reply:
Done.